api/controllers/linux/install/controller.go

```diff
@@ -265,6 +268,24 @@ func NewInstallController(opts ...InstallControllerOption) (*InstallController,
 			appcontroller.WithAirgapBundle(controller.airgapBundle),
 			appcontroller.WithPrivateCACertConfigMapName(adminconsole.PrivateCASConfigMapName), // Linux installations use the ConfigMap
 		)
+
+		// Detect registry settings and pass to app controller
+		var license *kotsv1beta1.License
+		if controller.license != nil {
+			// Parse license from bytes
+			var parsedLicense kotsv1beta1.License
+			if err := yaml.Unmarshal(controller.license, &parsedLicense); err != nil {
+				return nil, fmt.Errorf("failed to parse license data: %w", err)
+			}
+			license = &parsedLicense
+		}
+		registrySettings, err := controller.detectRegistrySettings(license)
```

Comment:
this happens when the api starts, which is before the user goes through the setup page, so if the user changes the cidr then the registry settings would have the wrong values